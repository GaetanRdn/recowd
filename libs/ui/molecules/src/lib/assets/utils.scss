@function map-get-deep($map, $level, $subLevel) {
  @return map-get($map, $level, $subLevel);
}

@mixin use-typography($typography) {
  @each $key in font-size, font-family, font-weight, line-height {
    #{$key}: map-get($typography, $key);
  }
}
