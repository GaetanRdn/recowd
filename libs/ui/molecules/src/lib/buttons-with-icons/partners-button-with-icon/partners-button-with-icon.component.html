<a rcButton role="button">
  <rc-partner-icon></rc-partner-icon>
  {{ 'partners' | translate }}
</a>
