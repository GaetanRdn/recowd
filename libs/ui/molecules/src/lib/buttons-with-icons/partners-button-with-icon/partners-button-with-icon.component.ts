import { ChangeDetectionStrategy, Component } from '@angular/core';
import { ButtonDirective } from '@recowd/ui-atoms';
import { TranslateModule } from '@ngx-translate/core';
import { PartnerIconComponent } from '../../material-icons/partner-icon/partner-icon.component';

@Component({
  selector: 'rc-partners-button-with-icon',
  standalone: true,
  imports: [ButtonDirective, PartnerIconComponent, TranslateModule],
  template: `<a rcButton role="button">
    <rc-partner-icon></rc-partner-icon>
    {{ 'partners' | translate }}
  </a>`,
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class PartnersButtonWithIconComponent {}
