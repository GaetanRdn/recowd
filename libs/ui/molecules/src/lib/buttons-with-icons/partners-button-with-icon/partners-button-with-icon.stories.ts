import { Meta, moduleMetadata, Story } from '@storybook/angular';
import { PartnersButtonWithIconComponent } from './partners-button-with-icon.component';
import { TranslateModule } from '@ngx-translate/core';

export default {
  title: 'molecules/buttons-with-icons/partners-button-with-icon',
  decorators: [
    moduleMetadata({
      imports: [PartnersButtonWithIconComponent, TranslateModule],
    }),
  ],
} as Meta<PartnersButtonWithIconComponent>;

const Template: Story<PartnersButtonWithIconComponent> = () => ({
  template: `<rc-partners-button-with-icon></rc-partners-button-with-icon>`,
});

export const Default = Template.bind({});
