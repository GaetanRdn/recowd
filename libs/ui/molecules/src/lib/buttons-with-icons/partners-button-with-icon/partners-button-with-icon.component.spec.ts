import { TestBed, waitForAsync } from '@angular/core/testing';

import { PartnersButtonWithIconComponent } from './partners-button-with-icon.component';
import { Component } from '@angular/core';
import { TemplateLookup } from '@recowd/test/utils';
import { TranslateModule } from '@ngx-translate/core';

describe('PartnersButtonWithIconComponent', () => {
  let templateLookup: TemplateLookup<HostComponent>;

  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      declarations: [HostComponent],
      imports: [PartnersButtonWithIconComponent, TranslateModule.forRoot()],
    }).compileComponents();

    templateLookup = new TemplateLookup(HostComponent);
    templateLookup.detectChanges();
  }));

  it('should create', () =>
    expect(templateLookup.firstChildElement).toMatchSnapshot());
});

@Component({
  template: ` <rc-partners-button-with-icon></rc-partners-button-with-icon>`,
})
export class HostComponent {}
