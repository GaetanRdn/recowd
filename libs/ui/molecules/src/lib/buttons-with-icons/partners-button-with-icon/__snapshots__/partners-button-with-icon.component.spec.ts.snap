// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PartnersButtonWithIconComponent should create 1`] = `
<rc-partners-button-with-icon>
  <a
    class="rc-button"
    rcbutton=""
    role="button"
  >
    <rc-material-icon
      type="outlined"
    >
      school
    </rc-material-icon>
     partners

  </a>
</rc-partners-button-with-icon>
`;
